language-support/ts/codegen/src/TsCodeGenMain.hs

```diff
@@ -315,3 +362,34 @@ onHead :: (a -> a) -> [a] -> [a]
 onHead f = \case
     [] -> []
     x:xs -> f x:xs
+
+onLast :: (a -> a) -> [a] -> [a]
+onLast f = \case
+  [] -> []
+  [l] -> [f l]
+  x : xs -> x : onLast f xs
+
+-- A type such as
+--   data Q = C { x: Int, y: Text }| G { z: Bool }
+-- has a DAML-LF representation like,
+--   record Q.C = { x: Int, y: String }
+--   record Q.G = { z: Bool }
+--   variant Q = C Q.C | G Q.G
+-- Use this function to get the definitions of 'Q.C' and 'Q.G' given
+-- 'Q'.
+assocDefDataTypes :: Module -> T.Text -> [((T.Text, T.Text), DefDataType)]
```

Comment:
This function is so small, I think it would increase readability if we inlined it, including the comment.

Reply:
Ok.

Reply:
Well, again, two call sites. I'll move it to the `where` clause as planned for `serFromDef`.